hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -470,8 +471,8 @@ writeLog x =
 --   combinators.
 --
 failWith :: (MonadTest m, HasCallStack) => Maybe Diff -> String -> m a
-failWith diff msg =
-  liftTest $ mkTest (Left $ Failure (getCaller callStack) msg diff, [])
+failWith mdiff msg =
```

Comment:
Do we care to pick up a better name instead of `mdiff`? Although I think we do similar in other places as well so probably not a huge deal.

Reply:
I tent to use an `m` prefix for maybe things is all. So when you peel it off it looks like this:
```haskell
foo mx =
  case mx of
    Nothing ->
      ..
    Just x ->
      ..
```